Sources/ClientRuntime/Networking/Http/URLSession/URLSessionHTTPClient.swift

```diff
@@ -330,4 +425,71 @@ public enum URLSessionHTTPClientError: Error {
     case unresumedConnection
 }
 
+extension Bundle {
+    func certificate(named name: String) -> SecCertificate? {
+        guard let cerURL = self.url(forResource: name, withExtension: "cer"),
+              let cerData = try? Data(contentsOf: cerURL) else {
+            return nil
+        }
+        return SecCertificateCreateWithData(nil, cerData as CFData)
+    }
+
+    func identity(named name: String, password: String) -> SecIdentity? {
+        guard let p12URL = self.url(forResource: name, withExtension: "p12"),
+              let p12Data = try? Data(contentsOf: p12URL) else {
+            return nil
+        }
+
+        let options = [kSecImportExportPassphrase as String: password] as CFDictionary
+        var items: CFArray?
+        let status = SecPKCS12Import(p12Data as CFData, options, &items)
+
+        guard status == errSecSuccess, let itemsArray = items as? [[String: AnyObject]],
+              let firstItem = itemsArray.first,
+              let identity = firstItem[kSecImportItemIdentity as String] else {
+            return nil
+        }
+
+        // Explanation of cross-platform behavior differences:
+        // - Linux and macOS treat Core Foundation types differently.
+        // - The `as? SecIdentity` casting causes a compiler error on Apple platforms as the cast is guaranteed.
+        // - Directly returning `identity` works on Linux but not on macOS due to strict type expectations.
+        // SwiftLint is temporarily disabled for the next line to allow a force cast, acknowledging the platform-specific behavior.
+        // swiftlint:disable:next force_cast
```

Comment:
Eventually Swift Foundation will eliminate these differences, but for now we're stuck with `swift-corelibs-foundation` on Linux which is not always faithful to the original.